File: core/src/com/jediterm/terminal/CursorShape.java
package com.jediterm.terminal;

/**
 * Cursor shape as described by <a href="https://vt100.net/docs/vt510-rm/DECSCUSR.html">DECSCUSR</a>.
 */
public enum CursorShape {
  BLINK_BLOCK,
  STEADY_BLOCK,
  BLINK_UNDERLINE,
  STEADY_UNDERLINE,
  BLINK_VERTICAL_BAR,
  STEADY_VERTICAL_BAR;

  public boolean isBlinking() {
    return this == BLINK_BLOCK || this == BLINK_UNDERLINE || this == BLINK_VERTICAL_BAR;
  }
}

File: jediterm-core-mpp/src/jvmMain/kotlin/com/jediterm/terminal/CursorShape.kt
package com.jediterm.terminal

/**
 * Cursor shape as described by [DECSCUSR](https://vt100.net/docs/vt510-rm/DECSCUSR.html).
 */
enum class CursorShape {
    BLINK_BLOCK,
    STEADY_BLOCK,
    BLINK_UNDERLINE,
    STEADY_UNDERLINE,
    BLINK_VERTICAL_BAR,
    STEADY_VERTICAL_BAR;

    val isBlinking: Boolean
        get() = this == CursorShape.BLINK_BLOCK || this == CursorShape.BLINK_UNDERLINE || this == CursorShape.BLINK_VERTICAL_BAR
}
